lib/rtp_h264/nal_formats/fu.ex

```diff
@@ -34,6 +34,46 @@ defmodule Membrane.Element.RTP.H264.FU do
     end
   end
 
+  @doc """
+  Fragmentate H264 unit into list of FU-A payloads
+  """
+  @spec fragmentate(binary(), pos_integer()) :: list(binary()) | {:error, :unit_too_small}
+  def fragmentate(data, preferred_size) do
+    with <<header::8, head::binary-size(preferred_size), rest::binary>> <- data,
+         <<r::1, nri::2, type::5>> <- <<header>> do
```

Comment:
this will always match. Move this to with body and then change `with` into a `case` since it's a `with`